java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/MoveRefactoring.java

```diff
@@ -0,0 +1,282 @@
+import org.netbeans.api.project.SourceGroup;
+import org.netbeans.modules.java.lsp.server.Utils;
+import org.netbeans.modules.parsing.api.ResultIterator;
+import org.netbeans.modules.refactoring.java.api.JavaMoveMembersProperties;
+import org.netbeans.modules.refactoring.java.api.JavaRefactoringUtils;
+import org.netbeans.spi.java.classpath.support.ClassPathSupport;
+import org.openide.filesystems.FileObject;
+import org.openide.util.NbBundle;
+import org.openide.util.lookup.Lookups;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Dusan Balek
+ */
+@ServiceProvider(service = CodeActionsProvider.class, position = 160)
+public final class MoveRefactoring extends CodeRefactoring {
+
+    private static final String MOVE_REFACTORING_KIND = "refactor.move";
+    private static final String MOVE_REFACTORING_COMMAND =  "java.refactor.move";
+    private static final ClassPath EMPTY_PATH = ClassPathSupport.createClassPath(new URL[0]);
+
+    private final Set<String> commands = Collections.singleton(MOVE_REFACTORING_COMMAND);
+    private final Gson gson = new Gson();
+
+    @Override
+    @NbBundle.Messages({
+        "DN_Move=Move...",
+    })
+    public List<CodeAction> getCodeActions(ResultIterator resultIterator, CodeActionParams params) throws Exception {
+        List<String> only = params.getContext().getOnly();
+        if (only == null || !only.contains(CodeActionKind.Refactor)) {
+            return Collections.emptyList();
+        }
+        CompilationController info = CompilationController.get(resultIterator.getParserResult());
+        if (info == null || !JavaRefactoringUtils.isRefactorable(info.getFileObject())) {
+            return Collections.emptyList();
+        }
+        info.toPhase(JavaSource.Phase.ELEMENTS_RESOLVED);
+        int offset = getOffset(info, params.getRange().getStart());
+        String uri = Utils.toUri(info.getFileObject());
+        Element element = elementForOffset(info, offset);
+        if (element != null) {
+            QuickPickItem elementItem = new QuickPickItem(createLabel(info, element));
+            elementItem.setUserData(new ElementData(element));
+            return Collections.singletonList(createCodeAction(Bundle.DN_Move(), MOVE_REFACTORING_KIND, null, MOVE_REFACTORING_COMMAND, uri, elementItem));
+        } else {
+            return Collections.singletonList(createCodeAction(Bundle.DN_Move(), MOVE_REFACTORING_KIND, null, MOVE_REFACTORING_COMMAND, uri));
+        }
+    }
+
+    @Override
+    public Set<String> getCommands() {
+        return commands;
+    }
+
+    @Override
+    @NbBundle.Messages({
+        "DN_DefaultPackage=<default package>",
+        "DN_SelectTargetPackage=Select target package",
+        "DN_CreateNewClass=<create new class>",
+        "DN_SelectTargetClass=Select target class",
+    })
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.